pkg/testing/fixture.go

```diff
@@ -164,15 +175,17 @@ func (f *Fixture) Prepare(ctx context.Context, components ...UsableComponent) er
 	return nil
 }
 
+// Run runs the Elastic Agent.
+//
+// If `states` are provided then the Elastic Agent runs until each state has been reached. Once reached the
+// Elastic Agent is stopped. If at any time the Elastic Agent logs an error log and the Fixture is not started
+// with `WithAllowErrors()` then `Run` will exit early and return the logged error.
+//
+// If no `states` are provided then the Elastic Agent runs until the context is cancelled.
 func (f *Fixture) Run(ctx context.Context, states ...State) error {
```

Comment:
Can we add an option for the fixture to run through the states and then leave the agent alive? 
It could be useful for the cases where we want the agent to reach a certain state before starting our tests.

Reply:
I do like that option. I will add that ability but I will do that in a follow up PR.